### PR TITLE
Update Joomla! to use latest MariaDB chart version

### PR DESCRIPTION
**Pull request by beltran-rubo**
**What this PR does / why we need it**: Update Joomla! to use the latest MariaDB chart version as dependency.

### CODE REVIEW

File: stable/joomla/Chart.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: joomla
-version: 1.0.3
+version: 1.0.4
```
Comment (carrodher) [inline stable/joomla/Chart.yaml]: @tompizmor implemented similar changes for `moodle` and `prestashop`. @sameersbn includes some comments about that https://github.com/kubernetes/charts/pull/5733/files#r190521131

I think we should have a consensus on this and apply it in the same way to the different charts.

File: stable/joomla/requirements.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 dependencies:
 - name: mariadb
-  version: 0.7.0
+  version: 4.1.0
```
Comment (carrodher) [inline stable/joomla/requirements.yaml]: Similar to the previous comment, take a look to @sameersbn comment https://github.com/kubernetes/charts/pull/5733/files#r190520117

Also, why are we using here a hardcoded version (`4.1.0`) while in the other PRs we are using other values?
```
-  version: 0.7.0
+  version: 3.x.x
```

https://github.com/kubernetes/charts/pull/5733/
https://github.com/kubernetes/charts/pull/5735/

File: stable/joomla/values.yaml [hunk 3]
```diff
@@ -72,42 +72,42 @@ externalDatabase:
 mariadb:
```
Comment (carrodher) [inline stable/joomla/values.yaml:72]: Can you reflect these changes in the README table and below commands?

https://github.com/kubernetes/charts/tree/master/stable/joomla#configuration

File: stable/joomla/Chart.yaml (hunk 1 above)
Comment (beltran-rubo) [reply stable/joomla/Chart.yaml]: Increased the minor version

File: stable/joomla/requirements.yaml (hunk 2 above)
Comment (beltran-rubo) [reply stable/joomla/requirements.yaml]: Moved to 4.x.x

File: stable/joomla/values.yaml (hunk 3 above)
Comment (beltran-rubo) [reply stable/joomla/values.yaml:72]: done

**Pull request merged**